src/app/zap-templates/templates/app/CHIPClientCallbacks-src.zapt

```diff
@@ -791,6 +760,86 @@ bool emberAfDiscoverCommandsReceivedResponseCallback(ClusterId clusterId, uint16
     return true;
 }
 
+static EmberAfStatus PrepareListFromTLV(TLV::TLVReader * tlvData, const uint8_t *& message, uint16_t & messageLen)
+{
+    CHIP_ERROR tlvError = CHIP_NO_ERROR;
+    TLV::TLVReader reader;
+    TLV::TLVType type;
+    reader.Init(*tlvData);
+    reader.EnterContainer(type);
+    tlvError = reader.Next();
+    if (tlvError != CHIP_NO_ERROR && tlvError != CHIP_END_OF_TLV)
+    {
+        return EMBER_ZCL_STATUS_INVALID_VALUE;
+    }
+    if (tlvError == CHIP_NO_ERROR)
+    {
+        tlvError   = reader.GetDataPtr(message);
```

Comment:
The TAG here is not important, since the ReadClient should hand finished the parse work (including checking for the tag), and we only need to care about the value here.

Reply:
would do this in https://github.com/project-chip/connectedhomeip/issues/7937